### PR TITLE
New README.md style

### PR DESCRIPTION
**Pull request by crystall1nedev**
Some changes to the README.md:

* Added table of contents to let people jump around easily
* Converted "Supported" section to a table, with new links to downloads
* Added 1.16.5 note for new installs of Dynmap on Spigot/PaperMC 1.16.5

### CODE REVIEW

**Comment by crystall1nedev** [comment]
Also, I've worked on parts of #3097 by adding a new GitHub Action that automatically check for any spelling errors, with support for custom wordlists. As of right now it only checks the markdown files of the repository but you can expand if you wish.

File: .github/workflows/spellcheck.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  push:
+
+jobs:
+  spellcheck:
+    name: rojopolis/spellcheck
+    runs-on: ubuntu-20.04
+    steps:
+    - name: actions/checkout
+      uses: actions/checkout@v2
+    - name: rojopolis/spellcheck - acutally doing something
```
Comment (rautamiekka) [inline .github/workflows/spellcheck.yaml]: Typo in `acutally`.

Comment (crystall1nedev) [reply .github/workflows/spellcheck.yaml]: lmao that was actually a joke to see if anyone would see the typo in the spellchecker. ahhh the irony of a spellchecker having to be spellchecked. i’ve corrected it though!

**Review by mikeprimm — APPROVED** [review]

**Comment by mikeprimm** [comment]
Looks interesting - let's give it a go.  Thanks for the contribution!

**Comment by crystall1nedev** [comment]
@mikeprimm Thanks! 

If in the future anyone needs to add a word to the custom word list they can add it to a new line in the .github/workflows/gibberish.txt file (and check the Actions tab for what was flagged), or if they want to add file types to scan for errors they can add it to the .spellcheck.yaml file.

**Pull request merged**